chromium_src/content/public/browser/web_ui_browser_interface_broker_registry.h

```diff
@@ -0,0 +1,65 @@
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_CHROMIUM_SRC_CONTENT_PUBLIC_BROWSER_WEB_UI_BROWSER_INTERFACE_BROKER_REGISTRY_H_
+#define BRAVE_CHROMIUM_SRC_CONTENT_PUBLIC_BROWSER_WEB_UI_BROWSER_INTERFACE_BROKER_REGISTRY_H_
+
+#include <vector>
+
+#include "base/functional/bind.h"
+#include "content/public/browser/per_web_ui_browser_interface_broker.h"
+#include "content/public/browser/render_frame_host.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/browser/web_ui.h"
+#include "mojo/public/cpp/bindings/binder_map.h"
+#include "mojo/public/cpp/bindings/pending_receiver.h"
+
+// Adds a method for exposing an interface to all WebUIs in the registry. Once
+// https://chromium-review.googlesource.com/c/chromium/src/+/7047465 lands
+// upstream we can remove this.
+#define AddBinderForTesting                                                 \
+  AddBinderForTesting_Unused() {}                                           \
+  template <typename Interface>                                             \
+  WebUIBrowserInterfaceBrokerRegistry& AddGlobal(                           \
+      base::RepeatingCallback<void(content::RenderFrameHost*,               \
+                                   mojo::PendingReceiver<Interface>)>       \
+          binder) {                                                         \
+    return AddGlobal<Interface>(base::BindRepeating(                        \
+        [](base::RepeatingCallback<void(content::RenderFrameHost*,          \
+                                        mojo::PendingReceiver<Interface>)>  \
+               binder,                                                      \
+           WebUIController* controller,                                     \
+           mojo::PendingReceiver<Interface> receiver) {                     \
+          binder.Run(controller->web_ui()->GetRenderFrameHost(),            \
+                     std::move(receiver));                                  \
+        },                                                                  \
+        std::move(binder)));                                                \
+  }                                                                         \
+  template <typename Interface>                                             \
+  WebUIBrowserInterfaceBrokerRegistry& AddGlobal(                           \
+      base::RepeatingCallback<void(                                         \
+          WebUIController*, mojo::PendingReceiver<Interface>)> binder) {    \
+    global_binder_initializers_.push_back(base::BindRepeating(              \
+        [](base::RepeatingCallback<void(                                    \
+               WebUIController*, mojo::PendingReceiver<Interface>)> binder, \
+           WebUIBinderMap* binder_map) {                                    \
+          binder_map->Add<Interface>(std::move(binder));                    \
```

Comment:
I know `BinderMapWithContext` defines behavior for duplicate registrations, I'd still add at least a `DCHECK` to avoid adding a binder for the same interface multiple times:
```cpp
DCHECK(!binder_map->Contains<Interface>());
```

Reply:
We want to be able to add a global binder which will duplicate some of upstreams (i.e. where they manually add it)